pods/pods.go

```diff
@@ -0,0 +1,213 @@
+// Package pods provides pod status for a namespace using kubectl.
+// It supports both a snapshot as well as a continuous stream.
+package pods
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+// A PodStatus represents the status of a single Pod.
+type PodStatus struct {
+	Name           string // name of pod
+	UID            types.UID
+	Namespace      string
+	Phase          corev1.PodPhase // current phase
+	Containers     []ContainerStatus
+	InitContainers []ContainerStatus
+	Pod            corev1.Pod // copy of the raw pod
+}
+
+func (p *PodStatus) String() string {
+	var buf strings.Builder
+	add := func(k, v string) {
+		if v != "" {
+			fmt.Fprintf(&buf, ", %s: %q", k, v)
+		}
+	}
+	fmt.Fprintf(&buf, "{Name: %q", p.Name)
+	add("UID", string(p.UID))
+	add("Namespace", p.Namespace)
+	add("Phase", string(p.Phase))
+	if len(p.Containers) > 0 {
+		fmt.Fprintf(&buf, ", Containers: {%s", p.Containers[0].String())
+		for _, c := range p.Containers[1:] {
+			fmt.Fprintf(&buf, ", %s", c.String())
+		}
+		fmt.Fprintf(&buf, "}")
+	}
+	if len(p.InitContainers) > 0 {
+		fmt.Fprintf(&buf, ", InitContainers: {%s", p.InitContainers[0].String())
+		for _, c := range p.InitContainers[1:] {
+			fmt.Fprintf(&buf, ", %s", c.String())
+		}
+		fmt.Fprintf(&buf, "}")
+	}
+	fmt.Fprintf(&buf, "}")
+	return buf.String()
+}
+
+func (p *PodStatus) Equal(q *PodStatus) bool {
+	if p.UID != q.UID ||
+		p.Name != q.Name ||
+		p.Namespace != q.Namespace ||
+		p.Phase != q.Phase ||
+		len(p.Containers) != len(q.Containers) ||
+		len(p.InitContainers) != len(q.InitContainers) {
+		return false
+	}
+
+	for i, c := range p.Containers {
+		if !c.Equal(&q.Containers[i]) {
+			return false
+		}
+	}
+
+	for i, c := range p.InitContainers {
+		if !c.Equal(&q.InitContainers[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// Values for Phase.  These constants are copied from k8s.io/api/core/v1 as a
+// convenience.
+const (
```

Comment:
these arent used anywhere

Reply:
No, they are available for the caller of the package so they don't have to go find the original ones.